src/plugins/template/backend/int_executable.cpp

```diff
@@ -83,14 +83,20 @@ class TemporaryOverrideOutputs {
         for (size_t i = 0; i < args.size(); ++i) {
             auto output = node->get_input_source_output(i);
             orig_shapes.push_back(output.get_partial_shape());
-            output.get_tensor().set_partial_shape(args[i].get_shape());
+            if (auto parameter = dynamic_cast<ov::op::v0::Parameter*>(node->get_input_node_ptr(i))) {
+                parameter->set_partial_shape(args[i].get_shape());
+                parameter->validate_and_infer_types();
+            }
```

Comment:
I assume this doesn't work -- we need to understand what shapes are being changed.